apps/aecore/lib/aecore/account/tx/coinbase_tx.ex

```diff
@@ -0,0 +1,140 @@
+defmodule Aecore.Account.Tx.CoinbaseTx do
+  @moduledoc """
+  Aecore structure of a transaction data.
+  """
+
+  @behaviour Aecore.Tx.Transaction
+
+  alias Aecore.Tx.DataTx
+  alias Aecore.Account.Tx.CoinbaseTx
+  alias Aecore.Account.Account
+  alias Aecore.Chain.ChainState
+  alias Aecore.Wallet.Worker, as: Wallet
+  alias Aecore.Account.Account
+  alias Aecore.Account.AccountStateTree
+
+  require Logger
+
+  @typedoc "Expected structure for the Spend Transaction"
+  @type payload :: %{
+          receiver: Wallet.pubkey(),
+          amount: non_neg_integer()
+        }
+
+  @typedoc "Reason for the error"
+  @type reason :: String.t()
+
+  @typedoc "Structure that holds specific transaction info in the chainstate.
+  In the case of SpendTx we don't have a subdomain chainstate."
+  @type tx_type_state() :: %{}
+
+  @typedoc "Structure of the Coinbase Transaction type"
+  @type t :: %CoinbaseTx{
+          receiver: Wallet.pubkey(),
+          amount: non_neg_integer()
+        }
+
+  @doc """
+  Definition of Aecore CoinbaseTx structure
+
+  ## Parameters
+  - receiver: To account is the public address of the account receiving coinbase tokens
+  - amount: The amount of tokens account should be granted
+  """
+  defstruct [:receiver, :amount]
+  use ExConstructor
+
+  # Callbacks
+
+  def get_chain_state_name, do: nil
+
+  @spec init(payload()) :: CoinbaseTx.t()
+  def init(%{receiver: receiver, amount: amount} = _payload) do
+    %CoinbaseTx{receiver: receiver, amount: amount}
+  end
+
+  @doc """
+  Creates a rewarding CoinbaseTx for the miner that mined the block
+  """
+  @spec create(binary(), integer()) :: payload()
+  def create(receiver, amount) do
+    %CoinbaseTx{receiver: receiver, amount: amount}
+  end
+
+  @doc """
+  Checks transactions internal contents validity
+  """
+  @spec is_valid?(CoinbaseTx.t(), DataTx.t()) :: boolean()
+  def is_valid?(%CoinbaseTx{amount: amount, receiver: receiver}, data_tx) do
+    cond do
+      amount < 0 ->
+        Logger.error("Value cannot be a negative number")
+        false
+
+      DataTx.fee(data_tx) != 0 ->
+        Logger.error("Fee has to be 0")
+        false
+
+      !Wallet.key_size_valid?(receiver) ->
+        Logger.error("Wrong receiver key size")
+        false
+
+      length(DataTx.senders(data_tx)) != 0 ->
```

Comment:
Shouldn't this be `!empty?`

Reply:
In other places you separate `senders = DataTx.senders(data_tx)`, better be consistent